File: benchmark/2026-09-17-7677581(latest)/semantic-compression/_superseded_run1/probes/python/F09.P1.py
def main() -> bool:
    s1 = "".join(["a", "b", "c"])
    s2 = "".join(["ab", "c"])
# BEGIN PROBE F09.P1
    eq = s1 == s2
# END PROBE F09.P1
    return eq


print(main())

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F09.P1.rs
fn main() {
    let base = String::from("ab");
    let s1 = base.clone() + "cd";
    let s2 = base + "cd";
// BEGIN PROBE F09.P1
    let eq = s1 == s2;
// END PROBE F09.P1
    println!("{} {}", eq, s1.as_ptr() != s2.as_ptr());
}
